python/understack-workflows/understack_workflows/main/undersync_device.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import os
+import sys
+
+from understack_workflows.helpers import boolean_args
+from understack_workflows.helpers import credential
+from understack_workflows.helpers import setup_logger
+from understack_workflows.nautobot import Nautobot
+from understack_workflows.undersync.client import Undersync
+
+logger = setup_logger(__name__)
+
+def update_nautobot(args) -> list[str]:
+    default_nb_url = "http://nautobot-default.nautobot.svc.cluster.local"
+    device_uuid = args.device_uuid
+    field_name = "connected_to_network"
+    field_value = args.network_name
+    nb_url = args.nautobot_url or default_nb_url
+
+    nb_token = args.nautobot_token or credential("nb-token", "token")
+    nautobot = Nautobot(nb_url, nb_token, logger=logger)
+    logger.info(
+        f"Updating Device {device_uuid} and moving it to '{field_value}' network."
+    )
+    nautobot.update_cf(device_uuid, field_name, field_value)
+    logger.debug(f"Updated Device.{field_name} to {field_value}")
+    switches = nautobot.uplink_switches(device_uuid)
+    logger.info(f"Obtained switch IDs: {switches}")
+    return switches
+
+
+def call_undersync(args, switches):
+    undersync_token = credential("undersync", "token")
+    if not undersync_token:
+        logger.error("Please provide auth token for Undersync.")
+        sys.exit(1)
+    undersync = Undersync(undersync_token)
+
+    try:
+        return undersync.sync_devices(switches, dry_run=args.dry_run, force=args.force)
+    except Exception as error:
+        logger.error(error)
+        sys.exit(2)
+
+
+def argument_parser():
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(__file__),
+        description="Trigger undersync run for a device",
+    )
+    parser.add_argument("--device_uuid", required=True, help="Nautobot device UUID")
```

Comment:
We can do this as a follow up but I think we should probably standardize this arg. The other entry points use `--device-id` and they use `type=UUID` to validate the input